### PR TITLE
Plotly exploration

### PR DESCRIPTION
**Pull request by kanastasakis26**
I added Plotly to the project. That code is limited to the DD dataviz type. So I only created a heatmap. You can find it in the 'plots' module.

I also made a few tweaks to that may or may not be improvements. 
The `Drawer` class is for opening and closing the drawer.
The `loadJson` tries to load JSON by alternating the position of the source and target names.
Yada, yada

### CODE REVIEW

File: client/ts/display_difference_chart.ts [hunk 1]
```diff
@@ -42,62 +44,136 @@ function clickHighlight(clickedElementList, nodes, sourceIndex, targetIndex, cli
...
+    clickHighlight(clickedElementList, nodes, sourceIndex, targetIndex, clickedElement);
+}
+
+
+/**
+ * Tries to load the JSON by alternating node names.
+ * It is not clear which name should be first from the graph.
+ * @param {string} sourceName
+ * @param {string} targetName
+ * @param onComplete
```
Comment (xyz8983) [inline client/ts/display_difference_chart.ts]: what is onComplete and onError? Are they customized call back function?

Comment (xyz8983) [reply client/ts/display_difference_chart.ts]: huh i think i see them  now

Comment (kanastasakis26) [reply client/ts/display_difference_chart.ts]: Added to documentation.

File: client/ts/drawer.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
+
+
+
+export class Drawer {
```
Comment (xyz8983) [inline client/ts/drawer.ts]: could you add some docstring to explain what the Drawer is doing?

Comment (kanastasakis26) [reply client/ts/drawer.ts]: No way.
Never gonna happen.

Comment (kanastasakis26) [reply client/ts/drawer.ts]: Okay... done.

Comment (xyz8983) [reply client/ts/drawer.ts]: 😈

File: client/ts/plots/heatmap.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    let y_label_index = getLabelIndexDict(y_labels);
+
+    let counts = create2dZeroMatrix(x_labels.length, y_labels.length);
+
+    let count_max = -99999;
+    let count_min = 99999;
+    let n = x.length;
+    for (let i = 0; i < n; i++){
+        counts[y_label_index[y[i]]][x_label_index[x[i]]] ++
+        count_max = Math.max(counts[y_label_index[y[i]]][x_label_index[x[i]]], count_max)
```
Comment (xyz8983) [inline client/ts/plots/heatmap.ts]: where are count_max and count_min used for? They seems not affecting the counts value

Comment (kanastasakis26) [reply client/ts/plots/heatmap.ts]: It might be needed when we want to manually set the color gradient.

Comment (kanastasakis26) [reply client/ts/plots/heatmap.ts]: I will comment them out for now.

Comment (kanastasakis26) [reply client/ts/plots/heatmap.ts]: done

**Pull request merged**